src/main/java/com/finance/app/infrastructute/configuration/ServiceConfig.java

```diff
@@ -0,0 +1,25 @@
+package com.finance.app.infrastructute.configuration;
+
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.mongodb.config.EnableMongoAuditing;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@EnableMongoAuditing
+public class ServiceConfig {
```

Comment:
why to you call it "ServiceConfig" while creating different kind of bins ? Just rename it properly and also try to separate bean definition by role